src/Gui/MainWindow.cpp

```diff
@@ -2509,6 +2516,33 @@ QMdiArea *MainWindow::getMdiArea() const
     return d->mdiArea;
 }
 
+void MainWindow::setWindowTitle(const QString& string)
+{
+    QString title;
+    QString appname =
+        QString(static_cast<QApplication*>(QCoreApplication::instance())->applicationName());
+    // allow to disable version number
+    ParameterGrp::handle hGen = +App::GetApplication().GetParameterGroupByPath(
+        "User parameter:BaseApp/Preferences/General");
+    bool showVersion = hGen->GetBool("ShowVersionInTitle", true);
+    if (showVersion) {
+        // set main window title with FreeCAD Version
+        auto config = App::Application::Config();
+        QString major = QString::fromLatin1(config["BuildVersionMajor"].c_str());
+        QString minor = QString::fromLatin1(config["BuildVersionMinor"].c_str());
+        QString point = QString::fromLatin1(config["BuildVersionPoint"].c_str());
+        QString suffix = QString::fromLatin1(config["BuildVersionSuffix"].c_str());
+        title = QString::fromLatin1("%1 %2.%3.%4%5").arg(appname, major, minor, point, suffix);
+    }
+    else {
+        title = appname;
+    }
+    if (!string.isEmpty()) {
+        title = QString::fromLatin1("%1 - %2").arg(string, title);
+    }
+    QMainWindow::setWindowTitle(title);
+}
+
```

Comment:
This function reads as a big blob of text - tray to add empty lines before  if statements, before last line and before `// allow ...` comment just like it was in the original code. It should be much easier on eyes.